cheri.hpp

```diff
@@ -49,10 +52,13 @@ struct CheriStackIterator
 
   }
   
-  // Skip all the things that arn't pointers.
-  // Also skip all the pointers we don't care about.  
+  // find the next valid GC pointer by iterating the stack and checking 
+  // the meta information for each 16 bytes.  
```

Comment:
Is it actually advancing 2 words at a time? I thought naively that it was advancing 1 word at a time to find non-CHERI pointers?

Reply:
Yes, it is advancing 16 bytes at a time. Side question isn't a word 16 bits, which makes double words 32 bits, quad words 64 bits and I suppose CHERI is using "octal words" or "double  quad words"  128 bits.

Reply:
A word on x86 is 8 bytes; I *assumed* (but I don't know!) that in CHERI a word is still 8 bytes, even though CHERI pointers are 16 bytes. I might be wrong.